rfcs/0018-glides-trainsheet-events.md

```diff
@@ -0,0 +1,571 @@
+- `departureTime` (Time, conditionally required): when the added trip is expected to depart `startLocation`.
+- `arrivalTime` (Time, conditionally required): when the added trip is expected to arrive at `endLocation`.
+- `previousTripKey` (Trip Key, optional): links the newly created trip to the trip immediately before it. If the Trip Key refers to an added trip, it may not have been seen in the event stream at the time of this message. Required if `endLocation` is specified and `arrivalTime` is NOT specified.
+- `nextTripKey` (Trip Key, optional): links the newly created trip to the trip after it. If the Trip Key refers to an added trip, it may not have been seen in the event stream at the time of this message. Required if `startLocation` is specified and `departureTime` is NOT specified.
+- `consist` (array of Optional Car, optional, default: empty list): the cars assigned to perform this trip. If a car is `null`, then no car is currently assigned to that position in the consist.
+- `operators` (list of Optional Operator, optional, default: empty list): the array of operators assigned to perform this trip. If an operator is `null`, then no operator is assigned to the car in that position in the consist.
+- `comment` (string, optional): free text with additional information about the trip
+
+*Notes*
+- At least one of `startLocation` and `endLocation` is required.
+- If `startLocation` is 
+- The event should only include values which were explicitly included by the `author`: inferred values should not be included.
+- `consist` and `operators` MUST be the same length; if they are not, the behavior of consumers is undefined.
+
+### Trip dropped
+This removes a trip (either added or scheduled), indicating that it will not be running.
+
+Event type: `com.mbta.ctd.glides.trip-dropped`
+Fields in the event:
+- `author` (Author): the inspector dropping the trip.
+- `tripKey` (Trip Key): which trip to drop. can be either an added or scheduled trip.
+- `reason` (string): a text description of why the trip is being dropped.
+
+### Trip restored
+This restores a trip (either added or scheduled) that was previously dropped.
+
+Event type: `com.mbta.ctd.glides.trip-restored`
+Fields in the event:
+- `author` (Author): the inspector restoring the trip.
+- `tripKey` (Trip Key): which trip to restore. can be either an added or scheduled trip.
+
+### Trip updated
+This indicates that a trip was updated in some fashion:  consist, operators, departure time. This includes updates which confirm an arrival or departure time. Fields which are not present are not considered to be updated.
+
+Event type: `com.mbta.ctd.glides.trip-updated`
+Fields in the event:
+- `author` (Author): the inspector adding the trip
+- `tripKey` (Trip Key): which trip is being updated
+- `comment` (string, optional): free text information about the trip
+- `route` (string, optional): if specified, the new route tag for the trip. This is different from a GTFS route ID.
```

Comment:
Rerouting trips is not currently supported in Glides and we don't expect it to be soon, so it's out of scope and shouldn't be here.

Reply:
That old comment for context:

> I would be in favor of having it in the spec. We haven't included or planned endpoint changes in the UI because it's not in paper trainsheets so we didn't need to do it to launch. But it's basic information about a trip, and our backend is set up to do it whenever we get around to designing the input for it.

And also this [comment](https://github.com/mbta/technology-docs/pull/18#discussion_r1089234791)

> We don't actually use this route field, and it can be removed. This field was added after a misunderstanding in a previous discussion. It appears on the paper trainsheets as a way to specify the start and end stations of a trip. But Glides doesn't use it, and if we ever report an end station of an added trip, it would be by the start/end location fields, not by a routeNumber field.

I still think it'd be nice to have it. However, I'd prefer listing start and end locations, rather than a `route`. The route numbers are just a harder-to-use encoding of the start and end locations.

Also, there's a couple discussions going around about `location` fields that would affect this, and if we want to delay figuring out how these modified locations tie in, that's okay.

Reply:
I think adding `endLocation` instead of `route` is fine, since that also covers crossovers (if that's added in the future). Is it possible to modify the `startLocation`?